### PR TITLE
OGM-1017 Store BigDecimal as decimal in Cassandra

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-1017

### CODE REVIEW

File: documentation/manual/src/main/asciidoc/en-US/modules/cassandra.asciidoc [hunk 1]
```diff
@@ -105,6 +105,7 @@ The type and field names can be overridden by annotations, but concatenation tok
 http://docs.datastax.com/en/cql/3.1/cql/cql_reference/cql_data_types_c.html[CQL3 types]
 
 CQL3 supports a smaller selection of numeric data types than Hibernate, so the missing types are autotmatically converted. byte and short values are promoted to integer.
```
Comment (gunnarmorling) [inline documentation/manual/src/main/asciidoc/en-US/modules/cassandra.asciidoc]: autotmatically=automatically

**Comment by gunnarmorling** [comment]
There should be a test for making sure that the mapped value is the one we expect. See `BuiltinTypeMappingTest` for some infra to use.

**Comment by DavideD** [comment]
Ah nice, I thought we didn't have those kind of tests in cassandra yet. I'll add it

**Comment by DavideD** [comment]
Am I wrong or the tests in that class are not really checking the type of the column? They are checking that the value returned is the expected one. how does it differ from the test in core?

**Comment by DavideD** [comment]
Ah no, right, I get it. What I was expecting is a test to check that a particular column is of type "decimal" in cassandra.

**Comment by DavideD** [comment]
Added two commits to fix the remarks

File: core/src/main/java/org/hibernate/ogm/type/impl/BigDecimalPassThroughType.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import java.math.BigDecimal;
+
+import org.hibernate.HibernateException;
+import org.hibernate.MappingException;
+import org.hibernate.engine.spi.Mapping;
+import org.hibernate.ogm.type.descriptor.impl.PassThroughGridTypeDescriptor;
+import org.hibernate.type.descriptor.java.BigDecimalTypeDescriptor;
+
+/**
+ * Type descriptor that will pass the {@link BigDecimal} to the dialect wihtout conversion.
```
Comment (gsmet) [inline core/src/main/java/org/hibernate/ogm/type/impl/BigDecimalPassThroughType.java]: s/wihtout/without/

File: documentation/manual/src/main/asciidoc/en-US/modules/cassandra.asciidoc [hunk 3]
```diff
@@ -104,7 +104,8 @@ The type and field names can be overridden by annotations, but concatenation tok
 
 http://docs.datastax.com/en/cql/3.1/cql/cql_reference/cql_data_types_c.html[CQL3 types]
 
-CQL3 supports a smaller selection of numeric data types than Hibernate, so the missing types are autotmatically converted. byte and short values are promoted to integer.
+CQL3 supports a smaller selection of numeric data types than Hibernate, so the missing types are automatically converted. byte and short values are promoted to integer.
```
Comment (gsmet) [inline documentation/manual/src/main/asciidoc/en-US/modules/cassandra.asciidoc]: The capitalization/style here is weird. I would have expected either Byte or `byte`.

**Comment by gunnarmorling** [comment]
Added a comment to address @gsmet's comments, rebased and applied. Thanks, @DavideD.

**Pull request closed without merging**